src/Orchard/Environment/DefaultOrchardHost.cs

```diff
@@ -343,6 +348,20 @@ public class DefaultOrchardHost : IOrchardHost, IShellSettingsManagerEventHandle
             _tenantsToRestart.GetState().Add(context.Settings);
         }
 
+        private void BlockRequestsDuringSetup() {
+            if (_shellContexts == null)
+                return;
+
+            // If there's only one tenant and it's initializing, return a Service Unavailable HTTP status code.

```

Comment:
OK this is fine for the "only one tenant" case, but what happens for other tenants while they are initializing?